[TGL][VTS] x86 vts_ltp_test fix

vdanix opened the pull request:
[TGL][VTS] x86 vts_ltp_test_x86 test fix

The ltp test cases are failing because of the below selinux violations.

avc: denied { write } for comm="init" name="swappiness" dev="proc" ino=22697
scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

avc: denied { write } for comm="init" name="disk_based_swap" dev="proc"
ino=22698 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

Added the permission to allow write for vendor_init.

Change-Id: Ibccae861f66715ea8c96ace406eff08e7e4a5bb6
Tracked-On: OAM-98169
Signed-off-by: vdanix <vishwanathx.dani@intel.com>

vdanix committed:
x86 vts_ltp_test_x86 test fix

The ltp test cases are failing because
of the below selinux violations

avc: denied { write } for comm="init"
name="swappiness" dev="proc" ino=22697
scontext=u:r:vendor_init:s0
tcontext=u:object_r:proc:s0 tclass=file permissive=0

Added the permission to allow write for vendor_init

Change-Id: Ibccae861f66715ea8c96ace406eff08e7e4a5bb6
Tracked-On: OAM-98169
Signed-off-by: vdanix <vishwanathx.dani@intel.com>

vdanix commented:
@sfonn could you please review this PR

vdanix commented:
@sundar.gnanasekaran@intel.com @swaroop.balan@intel.com could you please review this PR

vdanix commented:
@JeevakaPrabu could you please review this PR

JeevakaPrabu reviewed (changes requested):
Please update the commit message

<Short summary>

<Issue detailed>

<Fix detailed>

Tracked-On: OAM-98169
Signed-off-by: vdanix <vishwanathx.dani@intel.com>

JeevakaPrabu reviewed (commented):
@vdanix Zhenlong will share his comments

ceiba1985 commented on vendor/vendor_init.te:
```diff
@@ -2,3 +2,4 @@ allow vendor_init self:capability2 block_suspend;
 allow vendor_init kernel:system module_request;
 allow vendor_init debugfs_tracing_instances:dir create_dir_perms;
 allow vendor_init debugfs_tracing_instances:file w_file_perms;
+allow vendor_init proc:file write;
```
we can declare a new file label for swappiness and disk_based_swap:
https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/device/google/redbull-sepolicy/vendor/google/genfs_contexts#12
then add the following rule:
https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/device/google/redbull-sepolicy/vendor/google/vendor_init.te#6

patilgux replied:
Hi @Shao-Feng
Can u pls let us know what is the  sys entry for swappiness  for our Platform.
I have tried with /sys/vm/swappiness but i cant see this entry in device through adb shell
and device test cases are failing.

patilgux replied:
/proc/sys/vm/swappiness is this the proc file name which i can use in genfs_contexts

genfscon proc /proc/sys/vm/swappiness		 u:object_r:proc_swappiness:s0 like this?

ceiba1985 replied:
in file.te add:
type proc_swappiness, proc_type, fs_type;
in genfs_contexts add:
genfscon proc /sys/vm/swappiness                      u:object_r:proc_swappiness:s0
in vendor_inite.te add:
allow vendor_init proc_swappiness:file w_file_perms;

patilgux replied:
tried exactly the same way what you told.
Still the test cases are failing.

ceiba1985 replied:
do you add same rules for disk_based_swap?

patilgux replied:
no I have just added what mentioned for swappiness.
is different proc entry for disk_based_swap?
If I am not wrong you are asking to add the the same thing for

/proc/sys/vm/disk_based_swap?

ceiba1985 replied:
I only noticed these is an AVC error in the committed message:
avc: denied { write } for comm="init" name="disk_based_swap" dev="proc"
ino=22698 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

you should to figure out where is the disk_based_swap.

patilgux replied:
08-10 07:28:46.449   192   192 I auditd  : type=1400 audit(0.0:14): avc: denied { write } for comm="init" name="swappiness" dev="proc" ino=22697 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

08-10 07:28:46.449   192   192 I auditd  : type=1400 audit(0.0:15): avc: denied { write } for comm="init" name="disk_based_swap" dev="proc" ino=22698 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

08-10 07:28:46.453   392   392 I boot_progress_start: 7669

we have two denials and we kept only one otherwise commit message will be go longer.
So we need to add same way for disk_based_swap?

ceiba1985 replied:
yes, and please figure out the right path in proc for disk_based_swap.

rprajapx committed:
The ltp test cases are failing because

of the below selinux violations

avc: denied { write } for comm="init"
name="swappiness" dev="proc" ino=22697
scontext=u:r:vendor_init:s0
tcontext=u:object_r:proc:s0 tclass=file permissive=0

Added the permission to allow write for vendor_init
Tracked-On: OAM-98169
Signed-off-by: rprajapx <ravindrax.prajapati@intel.com>

Change-Id: I9117c4e9e7863ba03bea1344a06c5f641974320e

patilgux commented:
Hi @ceiba1985 
Please go through the changes and approve it.
Thanks for your inputs.
